src/util/strencodings.h

```diff
@@ -301,6 +301,39 @@ bool ConvertBits(O outfn, It it, It end, I infn = {}) {
     return true;
 }
 
+/** Convert from a set of power-of-2 number base to another set. */
+template<bool pad, typename O, typename It, typename I = IntIdentity>
+bool ConvertBitsVariable(std::vector<int> frombitsvec, std::vector<int> tobitsvec, O outfn, It it, It end, I infn = {}) {
+    if (frombitsvec.size() == 0 || tobitsvec.size() == 0) return false;
+    size_t fromindex = 0;
+    size_t toindex = 0;
+    auto frombits = [&frombitsvec, &fromindex] () { return fromindex >= frombitsvec.size() ? frombitsvec[frombitsvec.size()-1] : frombitsvec[fromindex];};
+    auto tobits = [&tobitsvec, &toindex] () { return toindex >= tobitsvec.size() ? tobitsvec[tobitsvec.size()-1] : tobitsvec[toindex];};
```

Comment:
In a3908f31e4dc48f87752791e5f04bb3c6006c168:

I'd avoid using lambdas for this. It makes the code much harder to follow. Easier to just define variables on each loop iteration and inline this computation.

Reply:
Fixed, Thanks!